pkg/postgres/pgadmin/admin_utils.go

```diff
@@ -492,3 +496,35 @@ func GetAllDatabases(postgresConfig *postgres.Config) ([]string, error) {
 
 	return clones, nil
 }
+
+// GetServerVersion - returns the version of PostgreSQL in numeric form.
+func GetServerVersion(postgresConfig *postgres.Config) (int64, error) {
```

Comment:
Why not passing context to this function?

Reply:
I'm not sure about rationale for that, but not a single function in pgadmin pack accepts a context so it would be consistent to keep it that way. I guess the idea is that they are so high privileged, that it does not make sense to fine tune it via context.

Reply:
> not a single function in pgadmin pack accepts a context
Which is not cool, right? The external context may not only have configured access scope, but also timeout and cancellation mechanism.

Reply:
Why do we need this method?  And why `server_version_num` vs `server_version`?  I think we could simply call https://github.com/stackrox/stackrox/blob/3ca0b326030110bb865b4aa3a14f4662c0123913/central/globaldb/postgres.go#L164 and be done with it.  We can get a reference to the db by calling https://github.com/stackrox/stackrox/blob/3ca0b326030110bb865b4aa3a14f4662c0123913/central/globaldb/postgres.go#L114.

If we absolutely need this method and need to do it this way then we should replace the other one with this one so we only get the version once.  Though I'm convinced using the existing method I referenced above is the simplest path.

Reply:
As for these methods not accepting a context their purpose doesn't dictate it.  SAC certainly doesn't apply to these and they are only used by migrator, backup/restore, and stats collections.  

If you disagree with that original implementation, feel free to write a ticket detailing the value passing a context would provide.

Reply:
Oh, didn't realize there is some existing functionality for that. Probably indicates that it could be placed better.

The difference between server_version and server_version_num is that the former may contain some textual parts, e.g. "18devel". But hopefully it's rare, so lets go with it.

Reply:
I would think we would want the textual version as it would probably be useful to know if someone was running "18devel".

Reply:
> As for these methods not accepting a context their purpose doesn't dictate it.

The telemetry gathering function, introduced in this PR, is called with a context, which is cancelled when telemetry collection is being stopped by a user, or on central shutdown. Feel free to ignore that.